crates/common/consensus/src/helpers.rs

```diff
@@ -98,3 +103,55 @@ pub fn get_voting_source(store: &Store, block_root: B256) -> Checkpoint {
         head_state.current_justified_checkpoint
     }
 }
+
+pub fn is_valid_indexed_attestation(
+    state: &BeaconState,
+    indexed_attestation: &IndexedAttestation,
+) -> bool {
+    let indices: Vec<usize> = indexed_attestation
+        .attesting_indices
+        .iter()
+        .map(|&i| i as usize)
+        .collect();
+
+    if indices.is_empty() || !is_sorted_and_unique(&indices) {
+        return false;
+    }
+
+    let pubkeys: Vec<&PubKey> = indices
+        .iter()
+        .map(|&i| &state.validators.get(i).unwrap().pubkey)
+        .collect();
+
+    let domain = match state.get_domain(
+        DOMAIN_BEACON_ATTESTER,
+        Some(indexed_attestation.data.target.epoch),
+    ) {
+        Ok(domain) => domain,
+        Err(_) => return false,
```

Comment:
```
/// Return the domain for the ``domain_type`` and ``fork_version``
pub fn compute_domain(
    domain_type: B32,
    fork_version: Option<B32>,
    genesis_validators_root: Option<B256>,
) -> anyhow::Result<B256> {
    let fork_data = ForkData {
        current_version: fork_version.unwrap_or_default(),
        genesis_validators_root: genesis_validators_root.unwrap_or_default(),
    };
    let fork_data_root = fork_data.compute_fork_data_root();
    let domain_bytes = [&domain_type.0, &fork_data_root.0[..28]].concat();
    Ok(B256::from_slice(&domain_bytes))
}
```

This function shouldn't return a result it should just return a B256, so remove the Result so we don't have to deal with it here